adit/upload/static/upload/upload.js

```diff
@@ -0,0 +1,361 @@
+    stopUploadVar: Boolean,
+    pbVisible: String,
+    uploadCompleteTextVisible: String,
+    stopUploadButtonStyleDisplay: String,
+
+    initUploadForm: function (destEl) {
+      document.body.addEventListener("chooseFolder", (e) => {
+        this.chooseFolder();
+      });
+      var button = formEl.querySelector("button#uploadButton");
+      this.stopUploadVar = false;
+      this.stopUploadButtonVisible = false;
+      this.fileCount = 0;
+      // Add an event listener to the button
+      button.addEventListener("click", function () {
+        // Trigger the form submission when the button is clicked
+        var myForm = formEl.querySelector("form#myForm");
+        if (myForm instanceof HTMLFormElement) {
+          htmx.trigger("#myForm", "submit");
+        }
+      });
+    },
+
+    initDestination: function (destEl) {
+      this._updateIsDestinationFolder(destEl);
+    },
+    onDestinationChange: function (ev) {
+      this._updateIsDestinationFolder(ev.target);
+      updatePreferences("upload", {
+        [UPLOAD_DESTINATION]: ev.target.value,
+      });
+    },
+    _updateIsDestinationFolder: function (destEl) {
+      const option = destEl.options[destEl.selectedIndex];
+      this.isDestinationFolder = option.dataset.node_type === "folder";
+    },
+
+    getFiles: function () {
+      var inputElement = formEl.querySelector("#fileselector");
+
+      if (
+        inputElement instanceof HTMLInputElement &&
+        inputElement.files.length > 0
+      ) {
+        return inputElement.files;
+      } else if (this.droppedFiles.length > 0) {
+        const files = [];
+        for (const f of this.droppedFiles) {
+          files.push(f);
+        }
+        return files;
+      } else {
+        return [];
+      }
+    },
+
+    toggleUploadButtonVisibility: function () {
+      // Check if files are selected
+      const files = this.getFiles();
+      this.buttonVisible = files.length > 0 ? true : false;
+      this.fileCount = files.length;
+      this.uploadCompleteTextVisible = false;
+    },
+    clearFiles: function () {
+      var inputEl = formEl.querySelector("#fileselector");
+
+      if (inputEl instanceof HTMLInputElement) {
+        inputEl.value = null;
+      }
+      this.droppedFiles = [];
+      this.toggleUploadButtonVisibility();
+    },
+
+    fileHandler: async function (fileObj, datasets) {
+      const arrayBuffer = await fileObj.arrayBuffer(); //await fileReader.readAsArrayBuffer(fileObj);
+      datasets.push(arrayBuffer);
+    },
+
+    chooseFolder: function () {
+      const files = this.getFiles();
+      this.loadFiles(files);
+    },
+
+    loadFiles: async function (files) {
+      const destinationSelect = formEl.querySelector('[name="destination"]');
+
+      if (destinationSelect instanceof HTMLSelectElement) {
+        const selectedOption =
+          destinationSelect.options[destinationSelect.selectedIndex];
+        var node_id = selectedOption.getAttribute("data-node_id");
```

Comment:
Use .dataset to access data attributes

Reply:
statdessen zugreofen über 'selectedOption.dataset.node_id' 
-> nochmal selecteOtpion nachlesen